flow/project.py

```diff
@@ -1020,27 +1082,38 @@ def _create_run_job_operations(self, entrypoint, default_directives, job,
         :rtype:
             Iterator[JobOperation]
         """
+        if isinstance(jobs, signac.contrib.job.Job):
+            jobs = [jobs]
+        else:
+            jobs = list(jobs)
+
+        project = jobs[0]._project
+        for job in jobs:
+            assert project == job._project
+
+        aggregated_jobs = self.aggregate(jobs)
         for name, op in self.operations.items():
-            if op.eligible(job, ignore_conditions):
-                directives = self._resolve_directives(name, default_directives, job)
-                cmd = self._run_cmd(entrypoint=entrypoint, operation_name=name,
-                                    operation=op, directives=directives, job=job)
-                # Uses a different id than the groups direct id. Do not use this for submitting
-                # jobs as current implementation prevents checking for resubmission in this case.
-                # The different ids allow for checking whether JobOperations created to run directly
-                # are different.
-                job_op = JobOperation(self._generate_id(job, name, index=index), name, job,
-                                      cmd=cmd, directives=deepcopy(directives))
-                # Get the prefix, and if it's not NULL, set the fork directive
-                # to True since we must launch a separate process. Override
-                # the command directly.
-                prefix = job._project._environment.get_prefix(job_op)
-                if prefix != '':
-                    job_op.directives['fork'] = True
-                    job_op._cmd = '{} {}'.format(prefix, job_op.cmd)
-                yield job_op
-
-    def _get_submission_directives(self, default_directives, job):
+            for aggregate in aggregated_jobs:
+                if op.eligible(*aggregate, ignore_conditions=ignore_conditions):
+                    directives = self._resolve_directives(name, default_directives, aggregate)
+                    cmd = self._run_cmd(entrypoint=entrypoint, operation_name=name,
+                                        operation=op, directives=directives, jobs=aggregate)
+                    # Uses a different id than the groups direct id. Do not use this for submitting
+                    # jobs as current implementation prevents checking for resubmission in this
+                    # case. The different ids allow for checking whether JobOperations created to
+                    # run directly are different.
```

Comment:
This isn't related, but I don't think this is true any more.

Reply:
Should I make a separate PR for this?

Reply:
It may be worth doing that, and we could just merge right away. No need for a changelog even.

Reply:
Addressed in #327. I will merge that PR.